ament_cmake_pep257/doc/index.rst

```diff
@@ -0,0 +1,42 @@
+ament_pep257
+============
+
+Checks the code style of Python source files using `pep257
+<http://pep257.readthedocs.org/>`_.
+Files with the following extensions are being considered: ``.py``.
+
+
+How to run the check from the command line?
+-------------------------------------------
+
+The command line tool is provided by the package `ament_pep257
+<https://github.com/ament/ament_lint>`_.
+
+
+How to run the check from within a CMake ament package as part of the tests?
+----------------------------------------------------------------------------
+
+``package.xml``:
+
+.. code:: xml
+
+    <build_depend>ament_cmake_test</build_depend>
```

Comment:
Well seeing as `ament_cmake` and `ament_cmake_pep257` both transitively depend on `ament_cmake_test`, does this need to be in there at all?

I'd argue that `ament_cmake_test` should just be in `ament_cmake_core` anyways, but I'd also say that about the majority of the packages in the `ament_cmake` repository, so I guess I'm arguing a failing point...

This is what I'd propose:

- `ament_cmake` depends on `ament_cmake_test` (already true)
- `ament_cmake` drops dependencies on things like `ament_cmake_{nose, gtest, gmock, ...}`
- we recommend users depend on `ament_cmake` and therefore the option `AMENT_ENABLE_TESTING` is always available
- the documentation for test suites and linters tell users to buildtool depend on `ament_cmake` and test depend on the package for the test suite or linter.

This way the `AMENT_ENABLE_TESTING` option is always there (listed in the cmake gui for example) but other test related packages are never brought in or `find_package`'d unless it is set externally.

Currently the documentation recommends:

```xml
<buildtool_depend>ament_cmake_core</buildtool_depend> <!-- was implied but not stated -->
<build_depend>ament_cmake_test</build_depend>
<test_depend>ament_cmake_pep257</test_depend>
```

```cmake
find_package(ament_cmake_core REQUIRED)  # again implied but not stated
find_package(ament_cmake_test REQUIRED)
if(AMENT_ENABLE_TESTING)
  find_package(ament_cmake_pep257 REQUIRED)
  ament_pep257()
endif()
```

But with the change I suggested:

```xml
<buildtool_depend>ament_cmake</buildtool_depend>
<test_depend>ament_cmake_pep257</test_depend>
```

```cmake
find_package(ament_cmake REQUIRED)
if(AMENT_ENABLE_TESTING)
  find_package(ament_cmake_pep257 REQUIRED)
  ament_pep257()
endif()
```

I think this is a more obvious expected usage and it avoids `ament_cmake` implicitly pulling in testing stuff like support for gtest and nosetest.

Do you see any issues with that?